processing/src/main/java/io/druid/query/search/search/SearchQuerySpec.java

```diff
@@ -29,11 +29,27 @@
     @JsonSubTypes.Type(name = "contains", value = ContainsSearchQuerySpec.class),
     @JsonSubTypes.Type(name = "insensitive_contains", value = InsensitiveContainsSearchQuerySpec.class),
     @JsonSubTypes.Type(name = "fragment", value = FragmentSearchQuerySpec.class),
-    @JsonSubTypes.Type(name = "regex", value = RegexSearchQuerySpec.class)
+    @JsonSubTypes.Type(name = "regex", value = RegexSearchQuerySpec.class),
+    @JsonSubTypes.Type(name = "all", value = SearchQuerySpec.TakeAll.class)
 })
 public interface SearchQuerySpec
 {
   public boolean accept(String dimVal);
 
   public byte[] getCacheKey();
+
+  class TakeAll implements SearchQuerySpec
```

Comment:
Even though this is simple, I'd prefer it being called `AllSearchQuerySpec` and being in its own file for consistency. please also define hashCode, equals, toString.